lib/collection/url.js

```diff
@@ -307,7 +307,7 @@ _.extend(Url, /** @lends Url */ {
         }
 
         // extract the query string
-        p.query = _.get(url.match(/^\?([^#$]+)/), '[1]');
+        p.query = _.get(url.match(/^\?([^#]+)/), '[1]');
```

Comment:
Yes. But seems not so correct now that I see it. There is a + operator. Can we add the failing test case and then the passing one?